### DISCUSSION TITLE
stately-xstate.edit not found

### DISCUSSION THREAD

**Discussion by butlerpaul** [statelyai/xstate-tools#343] [Q&A]
After fresh install I get "Command 'XState: Open Visual Editor' resulted in an error. (command stately-xstate.edit not found)"

Is there some dependency I need to install?

**Comment by davidkpiano** [comment]
That's strange; what extension version and VS Code version do you have? I'm unable to reproduce this.

**Comment by butlerpaul** [comment]
VSCode 1.68.1
Ubuntu 22.04 and 20.04 - same error
xstate vscode extension 1.8.3

**Comment by butlerpaul** [comment]
Are you able to reproduce?  I have updated VSCode to 1.69 and still see this issue.

**Reply by davidkpiano** [reply]
Unfortunately, I am not. I am on macos though, so I will have to try on Ubuntu.

**Reply by Andarist** [reply]
@butlerpaul would you be open to scheduling a call or something to go through this problem together in an attempt to gather more information? We've tried to look into it and reproduce but we couldn't do it so far

**Comment by butlerpaul** [comment]
Sure. Happy to help.

**Reply by butlerpaul** [reply]
Git it working.... pilot error!

**Reply by Andarist** [reply]
Do you know what was wrong with your setup?

**Comment by jean-smaug** [comment]
👋  Hi, I'm reproducing on MacOS
It's a fresh install of the extension.

```
Version : 1.69.1
Validation : b06ae3b2d2dbfe28bca3134cc6be65935cdfea6a
Date : 2022-07-12T08:21:51.333Z
Electron : 18.3.5
Chromium: 100.0.4896.160
Node.js : 16.13.2
V8 : 10.0.139.17-electron.0
Système d’exploitation : Darwin x64 21.5.0
```

**Reply by jean-smaug** [reply]
The problem was fixed after sign in.

**Reply by Andarist** [reply]
Does it happen again if you log out?

**Comment by davidkpiano** [comment]
If this is an authentication issue, we are removing the need for authenticating to use the extension soon.

**Comment by trystately** [comment]
7/25/2022 - also seeing this issue.  Here are the steps I took.

* Installed XState VSCode extension

   ```
   Name: XState VSCode
   Id: statelyai.stately-vscode 
   Description: Visual editing, autocomplete and typegen for XState
   Version: 1.8.3
   Publisher: statelyai
   VS Marketplace Link: https://open-vsx.org/vscode/item?itemName=statelyai.stately-vscode
   ```

* Created a simple `sign-in` machine on https://stately.ai 
* Exported the file to `machine.json`

```
   {
     "id": "sign-in",
     "initial": "init",
     "states": {
       "init": {
         "on": {
           "start": {
             "target": "initializing"
           }
         }
       },
       "initializing": {
         "on": {
           "success": {
             "target": "signing-in"
           },
           "failed": {
             "target": "failed"
           }
         }
       },
       "signing-in": {
         "on": {
           "success": {
             "target": "signed in"
           },
           "failed": {
             "target": "failed"
           }
         }
       },
       "failed": {},
       "signed in": {}
     }
   }
```   

* Opened file in VSCode
* Tried **XState: Open in Visual Editor** command.  Observed this error
 
  ![Screen Shot 2022-07-25 at 5 40 44 PM](https://user-images.githubusercontent.com/109999669/180898353-e2155a3b-b121-476a-8ad6-37a805a718f4.png)

* Tried **XState: Sign in to Stately** command.  Observed this error

   ![Screen Shot 2022-07-25 at 5 46 34 PM](https://user-images.githubusercontent.com/109999669/180898869-aa0ff2f4-4f1a-4a0a-97c4-f21ca29e2f4b.png)

For what it's worth I'm on macOS Catalina 10.15.7 
I created the original machine in Firefox 102.0.1 (64 bit)

**Comment by akrea** [comment]
Hi

Is there an update on this issue? Version 1.71.2 installed...

Thank you

**Reply by davidkpiano** [reply]
What version of the XState extension are you using?

You can find it from the extension page: 
![CleanShot 2022-09-18 at 10 41 43@2x](https://user-images.githubusercontent.com/1093738/190912780-061c7261-bda0-4bcc-aa90-5ff86685e7bf.png)

**Comment by WilkinsKa1** [comment]
Hello,

I'm also having this  issue when trying to use either the visualizer or inspector commands:
<img width="314" alt="Screen Shot 2022-09-21 at 11 31 29 AM" src="https://user-images.githubusercontent.com/2565527/191547320-db02bdd6-1a3e-4303-b32b-de93c01cf099.png">

VSCode version: 1.71.2 (Universal)
Xstate extension version: v1.10.0

**Reply by Andarist** [reply]
How do you attempt to trigger those commands? Do you see any errors in the VS Code developer tools (cmd+p -> "Developer: Toggle Developer Tools")?

**Comment by wohanley** [comment]
Same here. Linux, VS Code v1.71.2, XState extension v1.10.0. Opening a JSON file containing an XState machine definition then doing Ctrl+Shift+P ">XState: Open Visual Editor" gives me "Command 'XState: Open Visual Editor' resulted in an error. (command stately-xstate.edit not found)". Nothing in the VS Code developer tools console and nothing in any of the output logs as far as I can see, not that I'd know where to look. Happy to help debug.

**Reply by Andarist** [reply]
Would you be so kind to check out if the same thing happens if you install a local version of the extension, built from
this repo https://github.com/statelyai/xstate-tools ? You can start the extension there by using a launch script from here:
![image](https://user-images.githubusercontent.com/9800850/191936606-05f8c723-7a44-4a7f-adfd-d081ddbc8ea0.png)

Note that it might be best to remove the production extension first, before testing this out.

**Reply by wohanley** [reply]
Ah, I think I understand the problem now. All I had in my project was machine.json, but the extension activates for JS/TS files. So, I made a TS file with

```typescript
import { createMachine } from 'xstate';

const machine = createMachine(...);
```

just like the https://stately.ai/viz example, and it works. I misunderstood what the extension was expecting.

**Reply by akrea** [reply]
> Ah, I think I understand the problem now. All I had in my project was machine.json, but the extension activates for JS/TS files. So, I made a TS file with
> 
> ```ts
> import { createMachine } from 'xstate';
> 
> const machine = createMachine(...);
> ```
> 
> just like the https://stately.ai/viz example, and it works. I misunderstood what the extension was expecting.

I figured this out too! The machine has to be "declared" before stately starts working.

**Reply by Andarist** [reply]
> I figured this out too! The machine has to be "declared" before stately starts working.

Could you describe what do you mean in more detail?

**Reply by akrea** [reply]
Well: I expectetd the extension show an empty window where I could start inserting states, while it provides the code on the other half of the vertically split window. But in order to be able to "open" the xstate extension you need to provide the minimal code snippet in a file. Only then the extension works. Providing this code snippet is also the trigger when the "Open Visual Editor | Open Inspector" options are shown in the code window.

Hope this helps.

**Reply by Andarist** [reply]
Ah yes, this makes sense - we could have a "create new machine" command or something like that.

**Comment by yureckey** [comment]
I have this problem too.
Just installed xstate vscode extension (1.10.0) in Quasar generated app (Vue3)
![xstateerror](https://user-images.githubusercontent.com/535466/198882117-5384356c-aac9-446d-bee3-1076cef41476.png)

**Comment by glimmbo** [comment]
+1 With some errors being found in the `RUNTIME STATUS` tab of the extension page:
![Screenshot 2023-05-16 at 9 04 06 PM](https://github.com/statelyai/xstate/assets/29473352/6fb8e46b-cb8a-4ab2-a4d1-02762fd8b830)

**Reply by Andarist** [reply]
are you using tab-based indentation by any chance?
